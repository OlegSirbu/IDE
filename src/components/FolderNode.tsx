import React, { useState } from 'react';
import FileNode from './FileNode';
import { sortFiles } from '../utils/fileUtils';

const FolderNode: React.FC<FolderNodeProps> = ({
  name,
  children,
  selectedFilePath,
  onSelectFile
}) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  const toggleFolder = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div>
      <div onClick={toggleFolder} className="pointer">
        {isOpen ? '📂' : '📁'} {name}
      </div>
      {isOpen && (
        <div className="folder-children">
          {Object.entries(children)
            .sort(sortFiles)
            .map(([key, value]: [string, FileNodeStructure]) => {
              if (value.type === 'file') {
                return (
                  <FileNode
                    key={key}
                    isSelected={selectedFilePath === value.fullPath}
                    onSelect={onSelectFile}
                    name={key}
                    fullPath={value.fullPath}
                  />
                );
              } else {
                return (
                  <FolderNode
                    onSelectFile={onSelectFile}
                    selectedFilePath={selectedFilePath}
                    key={key}
                    name={key}
                    children={value.children}
                  />
                );
              }
            })}
        </div>
      )}
    </div>
  );
};

export default FolderNode;
