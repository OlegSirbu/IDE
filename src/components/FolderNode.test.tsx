import React from 'react';
import { render } from '@testing-library/react';
import FolderNode from './FolderNode';

describe('FolderNode', () => {
  it('renders correctly and matches snapshot', () => {
    const mockChildren = {
      'file1.txt': { type: 'file', fullPath: 'path/to/file1.txt' }
    };

    const { asFragment } = render(
      <FolderNode
        name="root"
        children={mockChildren}
        selectedFilePath=""
        onSelectFile={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
