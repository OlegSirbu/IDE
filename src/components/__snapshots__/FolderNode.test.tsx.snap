// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderNode renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="pointer"
    >
      📁 root
    </div>
  </div>
</DocumentFragment>
`;
